model/remote/list.go

```diff
@@ -0,0 +1,72 @@
+package remote
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/cozy/cozy-stack/model/instance"
+	"github.com/cozy/cozy-stack/pkg/config/config"
+	"github.com/cozy/httpcache"
+)
+
+var listClient = &http.Client{
+	Timeout:   20 * time.Second,
+	Transport: httpcache.NewMemoryCacheTransport(32),
+}
+
+// https://docs.github.com/en/rest/repos/contents#get-repository-content
+const listURL = "https://api.github.com/repos/cozy/cozy-doctypes/contents/"
+
+type listEntries struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+}
+
+// ListDoctypes returns the list of the known remote doctypes.
+func ListDoctypes(inst *instance.Instance) ([]string, error) {
+	var doctypes []string
+	if config.GetConfig().Doctypes == "" {
+		req, err := http.NewRequest(http.MethodGet, listURL, nil)
+		if err != nil {
+			return nil, err
+		}
+		res, err := listClient.Do(req)
+		if err != nil {
+			log.Warnf("cannot list doctypes: %s", err)
+			return nil, ErrNotFoundRemote
+		}
+		defer res.Body.Close()
+		var entries []listEntries
+		if err := json.NewDecoder(res.Body).Decode(&entries); err != nil {
+			return nil, err
+		}
+		for _, entry := range entries {
+			if entry.Type != "dir" {
+				continue
+			}
+			if entry.Name == "docs" || strings.HasPrefix(entry.Name, ".") {
+				continue
+			}
+			doctypes = append(doctypes, entry.Name)
+		}
```

Comment:
It seems risky to me to rely on the fact that all non docs, non hidden files in this repository are remote doctype definitions.
It might not be easy to change that today but what do you think about adding an extra check for the presence of at least two `.` (dots) in the name for example?

Reply:
We have a few remote doctypes with a name that contains only one dot (like `com.inokufu`).